lib/spack/docs/build_systems/luapackage.rst

```diff
@@ -0,0 +1,82 @@
+.. Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+   Spack Project Developers. See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+.. _luapackage:
+
+------------
+MesonPackage
+------------
+
+Much like Autotools and CMake, Meson is a build system.  But it is
+meant to be both fast and as user friendly as possible.  GNOME's goal
+is to port modules to use the Meson build system.
+
+^^^^^^
+Phases
+^^^^^^
+
+The ``LuaPackage`` base class comes with the following phases:
+
+#. ``preprocess`` - generate ninja files
+#. ``install`` - install the project
+
+By default, these phases run:
+
+.. code-block:: console
+
+   $ # preprocess is a noop by default
+   $ luarocks make .
+
+
+Any of these phases can be overridden in your package as necessary.
+
+^^^^^^^^^^^^^^^
+Important files
+^^^^^^^^^^^^^^^
+
+Packages that use the Lua/LuaRocks build system can be identified by the
+presence of a ``.*\.rockspec`` file. This file declares things
+like build instructions and dependencies.
+
+It is common for the rockspec file to list the lua version required in
```

Comment:
Can you add a 1-line example of this? This helps people know what text to search for in that file

Reply:
Sure, I'll use this one:

```lua
dependencies = {
   "lua >= 5.1",
}
```